feat(site): allow modifying task prompts for starting tasks

DanielleMaywood opened the pull request:
This PR adds the frontend implementation for modifying task prompts. Closes https://github.com/coder/internal/issues/1084

---

🤖 PR was initially written by Claude Sonnet 4.5 Thinking using [Coder Mux](https://github.com/coder/cmux) and then heavily modified by a human 👩

DanielleMaywood committed:
feat(site): allow modifying task prompt prior to start

mafredri reviewed (commented):
The flow seems correct to me.

Visually I think it'd be beneficial to not just show "you can edit the prompt while ...", but also show the actual prompt here. Re-reading the prompt while waiting may help spark an idea for the user to more likely use the feature.

I also ran into the dialogue open but disabled "cannot change for active task". But that's because my sample template is very fast. It felt a bit clunky but I have no suggestion for improvement.

I'll defer FE patterns review to Kayla.

aslilac commented on site/src/api/api.ts:
```diff
@@ -2692,6 +2692,16 @@ class ApiMethods {
+	): Promise<void> => {
+		await this.axios.patch(`/api/v2/tasks/${user}/${id}/input`, {
+			input,
+		} satisfies TypesGen.UpdateTaskInputRequest);
```
god I love the `satisfies` keyword. thank you for actually knowing how to use it correctly. 💕

DanielleMaywood committed:
chore: a few changes

- Move to using `formik` and a form
- Remove unneeded comment
- Use DialogClose component
- Use `useId` instead of hard-coded IDs

DanielleMaywood committed:
chore: few more changes + merge main into branch

DanielleMaywood committed:
chore: appease linter

DanielleMaywood committed:
test: fix tests

DanielleMaywood committed:
chore: only cancel job if not completed

DanielleMaywood committed:
fix: tests

DanielleMaywood committed:
chore: make "Edit prompt" an outline instead
